pkg/apis/configuration/v1alpha1/types.go

```diff
@@ -82,11 +84,16 @@ type SelectorField struct {
 	Value string `json:"value"`
 }
 
-// TracingSpec is the spec object in ConfigurationSpec
+// TracingSpec defines distributed tracing configuration
 type TracingSpec struct {
 	SamplingRate string `json:"samplingRate"`
 }
 
+// MetricSpec defines metrics configuration
+type MetricSpec struct {
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
@yaron2 what's the best way to set this value to true by default ? seems like kubebuilder:default= marker doesn't work.

Reply:
I think this should work, since the CRD defaulting feature is GA since 1.17:

https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#defaulting

Not sure how we achieve the same semantic when not running on K8s though.

Reply:
We can make it `Enabled *bool` and then treat it as `true` in code if it is `nil`. WDYT?

Reply:
> We can make it `Enabled *bool` and then treat it as `true` in code if it is `nil`. WDYT?

let me try it out.

Reply:
before deserialize json payload to object, if object has the value, then json.marshal use this as the default value. so if crd component doesn't have metricspec, then json.marshal will use the existing value in the object. 

I applied the change and added test. please review it.

Reply:
@yaron2 @tcnghia I found why the // + marker is not working now. We have not generate crd based on the marker...

I created the separate issue. https://github.com/dapr/dapr/issues/2355
Once we fixed it, we can use default value for crd property.

Reply:
Does this mean we support a minimum version of 1.17?

Reply:
@yaron2 we already use v1 crd, which means minimum version is 1.16. kubebuilder:default option is one of the v1 crd features. please correct me if I am wrong.

Reply:
Ah. nghia's comment ... let me double-check

Reply:
![image](https://user-images.githubusercontent.com/11863108/97668376-0ab79f80-1a3f-11eb-9d25-769f1a0d3c60.png)

in 1.16, defaulting feature is enabled as beta. but I would like to double-check using minikube and aks 1.16.

Reply:
it works with aks 1.16.13 and minikube 1.16.10.